jdaviz/configs/imviz/tests/test_catalogs.py

```diff
@@ -99,3 +101,47 @@ def test_plugin_image_with_result(self, imviz_helper):
         catalogs_plugin.vue_do_clear()
 
         assert not catalogs_plugin.results_available
+
+        # test loading from file
+        table = imviz_helper.app._catalog_source_table
```

Comment:
Would be nice if we can turn this test into a test without remote data. Either create this table on-the-fly or generate a small ECSV test data file in a new test outside of this remote data test class.

Reply:
the ecsv file is created on the fly, but I used the existing catalogs test to do so (to make sure that it gives the same results as the SDSS query used to create the ecsv file).  It looks like this particular test doesn't actually use remote data, but the entire test class is marked as requiring remote data though... maybe we should just mark the individual methods or split the class into those that need remote data and those that don't?

Reply:
Just make sure the test is not set up that it downloads remote data regardless of what you are doing. But yes, if a test does not really need remote data, should not mark it as such.

Reply:
To clarify, if you are building the table from a query that needs internet, it still counts as remote data.

Reply:
A separate offline test would be nice, IMHO, to test that:

* the plugin does not access internet when it does not have to, and
* it can read something other than reformatted SDSS query (get a current real JWST ESCV file, trim it down to 2 rows, modify the SkyCoord to match image if you have to)